### PR TITLE
Core: set the base href of the context in parseHTML

### PR DESCRIPTION
**Pull request by timmywil**
Fixes gh-2965

### CODE REVIEW

File: src/core/parseHTML.js [hunk 1]
```diff
@@ -21,14 +21,29 @@ jQuery.parseHTML = function( data, context, keepScripts ) {
...
+	var base, parsed, scripts;
+
+	if ( !context ) {
+
+		// Stop scripts or inline event handlers from being executed immediately
+		// by using document.implementation
+		if ( support.createHTMLDocument ) {
+			context = document.implementation.createHTMLDocument( "" );
+
+			// Support: Chrome <=46-48
```
Comment (gibson042) [inline src/core/parseHTML.js]: Spaces around the `-`.

File: src/core/parseHTML.js [hunk 2]
```diff
@@ -21,14 +21,29 @@ jQuery.parseHTML = function( data, context, keepScripts ) {
...
+	var base, parsed, scripts;
+
+	if ( !context ) {
+
+		// Stop scripts or inline event handlers from being executed immediately
+		// by using document.implementation
+		if ( support.createHTMLDocument ) {
+			context = document.implementation.createHTMLDocument( "" );
+
+			// Support: Chrome <=46 - 48
```
Comment (mgol) [inline src/core/parseHTML.js]: This is not just Chrome but also Firefox as it doesn't provide the resolved URL there. Also, we're really required to do it so I'm not sure if we can claim it's to support some browser as if it had a bug. It's just something we have to do to make it work as we want, the previous way was incorrect.

Comment (timmywil) [inline src/core/parseHTML.js]: I suppose that's true. So just remove the support comment?

Comment (mgol) [inline src/core/parseHTML.js]: Yeah, I think it's not needed here.

-- 
Michał Gołębiowski

**Pull request closed without merging**